### PR TITLE
fix: make JSON formatter respect the value of DisableTime

### PR DESCRIPTION
**Pull request by marco-m**
Plus tests for both the plain and JSON output.

Closes: #74
Signed-off-by: Marco Molteni <marco.molteni@mailbox.org>

### CODE REVIEW

File: intlogger.go [hunk 1]
```diff
@@ -412,8 +412,11 @@ func (l *intLogger) logJSON(t time.Time, name string, level Level, msg string, a
 
 func (l intLogger) jsonMapEntry(t time.Time, name string, level Level, msg string) map[string]interface{} {
 	vals := map[string]interface{}{
-		"@message":   msg,
-		"@timestamp": t.Format("2006-01-02T15:04:05.000000Z07:00"),
+		"@message": msg,
+	}
+	if len(l.timeFormat) > 0 {
+		// FIXME this doesn't respect LoggerOptions.TimeFormat
```
Comment (jefferai) [inline intlogger.go:418]: Seems like this should be fixed before this is merged?

Comment (marco-m) [reply intlogger.go:418]: To be clear: this is another pre-existing bug that I discovered while fixing DisableTime, so I annotated it. I didn't want to mud the waters of this PR. Are you OK with fixing both in the same PR ?

Comment (marco-m) [reply intlogger.go:418]: ping @jefferai in case you missed my question ?

Comment (dnephin) [reply intlogger.go:418]: I think it would be fine to fix both here. Thank you

Comment (jefferai) [reply intlogger.go:418]: Yep, fixing both here would be great!

Comment (marco-m) [reply intlogger.go:418]: Hello @jefferai, @dnephin 
Fixing the TimeFormat is more complicated than expected.
Problem is that the _default_ resolution of the timestamp changes between the plain and JSON format.

Plain has millisecond resolution:
https://github.com/hashicorp/go-hclog/blob/e59fd7e1110848012d4335fcd0d3bb88e0cdf9e7/intlogger.go#L26

JSON has microsecond resolution:
https://github.com/hashicorp/go-hclog/blob/e59fd7e1110848012d4335fcd0d3bb88e0cdf9e7/intlogger.go#L420

If we want backward compatibility, we need a conditional. At this point I have the feeling we would make something that is already confusing even more confusing: the fact that the _lenght_ of the time format is overloaded to encode the meaning of DisableTime:

https://github.com/hashicorp/go-hclog/blob/e59fd7e1110848012d4335fcd0d3bb88e0cdf9e7/intlogger.go#L119-L123

At this point I think it is better to merge this PR as-is and make a separate PR to address custom time format for JSON. For the new PR, you also need to decide if backward compatibility in time resolution is important or not.

Comment (marco-m) [reply intlogger.go:422]: I opened #81 to fix the remaining problems.

Comment (marco-m) [reply intlogger.go:422]: Hello @jefferai @dnephin
I did additional work following your request, I would appreciate a review when you have time.
Thanks.

**Pull request merged**